.buildkite/scripts/build_component.sh

```diff
@@ -29,16 +44,28 @@ set_hab_binary() {
         # runs that may take place on it.
         sudo hab pkg install "${hab_ident}"
         sudo hab pkg install "$(buildkite-agent meta-data get studio-version)"
-        declare -g hab_binary="/hab/pkgs/${hab_ident}/bin/hab"
+        hab_binary="/hab/pkgs/${hab_ident}/bin/hab"
         declare -g new_studio=1
     else
-        echo "Buildkite metadata NOT found; using previously-installed hab binary"
-        hab_binary="$(which hab)"
-        declare -g hab_binary
+        echo "Buildkite metadata NOT found; using previously-installed hab binary: $hab_binary"
     fi
+    declare -g hab_binary
     echo "--- :habicat: Using $(${hab_binary} --version)"
 }
 
+install_hab_kernel2_binary() {
```

Comment:
The comments above about this being a temporary step may be better attached to this function directly.